test/integration/acceptance/grantable_permissions_fixture.hpp

```diff
@@ -0,0 +1,295 @@
+      const shared_model::interface::types::AccountNameType &user,
+      const shared_model::crypto::Keypair &key,
+      const shared_model::interface::RolePermissionSet &perms,
+      const shared_model::interface::types::RoleIdType &role);
+
+  /**
+   * Creates two accounts with corresponding permission sets
+   * @param itf - initialized instance of test framework
+   * @param perm1 set of permissions for account #1
+   * @param perm2 set of permissions for account #2
+   * @return reference to ITF object with two transactions
+   */
+  integration_framework::IntegrationTestFramework &createTwoAccounts(
+      integration_framework::IntegrationTestFramework &itf,
+      const shared_model::interface::RolePermissionSet &perm1,
+      const shared_model::interface::RolePermissionSet &perm2);
+
+  /**
+   * Create a transaction such that the creator grants permittee a permission
+   * @param creator_account_name - first's account name without domain
+   * @param creator_key - the keypair to sign the transaction
+   * @param permitee_account_name - a name of permittee account (receives the
+   * grantable permission)
+   * @param grant_permission - grantable permisssion to be granted
+   * @return proto::Transaction
+   */
+  shared_model::proto::Transaction accountGrantToAccount(
+      const shared_model::interface::types::AccountNameType
+          &creator_account_name,
+      const shared_model::crypto::Keypair &creator_key,
+      const shared_model::interface::types::AccountNameType
+          &permitee_account_name,
+      const shared_model::interface::permissions::Grantable &grant_permission);
+
+  /**
+   * Forms a transaction such that creator of transaction revokes a permission
+   * from permittee
+   * @param creator_account_name - first's account name without domain
+   * @param creator_key - the keypair to sign the transaction
+   * @param permittee_account_name - a name of permittee account (lost the
+   * grantable permission)
+   * @param revoke_permission - grantable permission to be revoked
+   * @return proto::Transaction
+   */
+  shared_model::proto::Transaction accountRevokeFromAccount(
+      const shared_model::interface::types::AccountNameType
+          &creator_account_name,
+      const shared_model::crypto::Keypair &creator_key,
+      const shared_model::interface::types::AccountNameType
+          &permittee_account_name,
+      const shared_model::interface::permissions::Grantable &revoke_permission);
+
+  /**
+   * Forms a transaction that either adds or removes signatory of an account
+   * @param f Add or Remove signatory function
+   * @param permitee_account_name name of account which is granted permission
+   * @param permitee_key key of account which is granted permission
+   * @param account_name account name which has granted permission to permitee
+   * @return a transaction
+   */
+  shared_model::proto::Transaction permiteeModifySignatory(
+      TxBuilder (TxBuilder::*f)(
```

Comment:
Consider using template here. It will be clearer. Check out: https://godbolt.org/g/tNZ1Fh for an example. You can still have definition in cpp, if you explicitly instantiate it in cpp file